src/js/dqa.jsx

```diff
@@ -21,122 +15,212 @@ var SortTypes = {
         
 var DQA = React.createClass({
     propTypes: {
-        content: React.PropTypes.string,
-        width: React.PropTypes.number,
+        aggregated: React.PropTypes.bool,
         data: React.PropTypes.object,
+        dict: React.PropTypes.object,
     },
 
     getDefaultProps: function() {
         return {
-            content: '',
-            width: 500,
+            aggregated: false,
             data: {},
+            dict: {}
         };
     },
 
     getInitialState: function() {
         return {
+            columnWidths: {
+                code: width,
+                desc: 3*width,
+                status: 2*width,
+                sites: 3*width,
+            },
             isCollapsed: false,
-            rows: this._getRows(),
-            sortBy: null,
-            sortDir: null,
+            rows: {},
+            sortBy: 'code',
+            sortDir: SortTypes.ASC,
+            tableWidth: 500,
         };
     },
 
-    _getRows: function() {
+    _getRows: function(props, sortBy) {
```

Comment:
this is called from 2 places, componentWillReceiveProps and componentWillMount -- the former passes in nextProps (because a new set of properties has just been received and  this.props hasn't been updated yet) and the latter passes in this.props